borg/testsuite/archiver.py

```diff
@@ -488,6 +488,37 @@ def test_extract_include_exclude(self):
             self.cmd('extract', '--exclude-from=' + self.exclude_file_path, self.repository_location + '::test')
         self.assert_equal(sorted(os.listdir('output/input')), ['file1', 'file3'])
 
+    def test_extract_include_exclude_regex(self):
+        self.cmd('init', self.repository_location)
+        self.create_regular_file('file1', size=1024 * 80)
+        self.create_regular_file('file2', size=1024 * 80)
+        self.create_regular_file('file3', size=1024 * 80)
+        self.create_regular_file('file4', size=1024 * 80)
+
+        # Create with --exclude-regex
+        self.cmd('create', '--exclude-regex=input/file4$', self.repository_location + '::test', 'input')
+        with changedir('output'):
+            self.cmd('extract', self.repository_location + '::test')
+        self.assert_equal(sorted(os.listdir('output/input')), ['file1', 'file2', 'file3'])
+        shutil.rmtree('output/input')
+
+        # Extract with --exclude-regex
+        with changedir('output'):
+            self.cmd('extract', '--exclude-regex=file3+', self.repository_location + '::test')
+        self.assert_equal(sorted(os.listdir('output/input')), ['file1', 'file2'])
```

Comment:
> hmm, this doesn't actually show that 3+ would match more than one "3".

It indeed doesn't. That wasn't the intention, however. The intention was to write something which wouldn't match if interpreted as a shell glob pattern. Do you want me to update the test to use more complicated filenames?

> and why is file4 not in the output?

It's excluded when creating the archive (`--exclude-regex=input/file4$`).